frontend/src/pages/frontPage/FrontPage.jsx

```diff
@@ -1,16 +1,95 @@
+                        loading:false,
+                        error:false
+                    })
+                }
+            }catch(error){
+                setState({
+                    loading:false,
+                    error:true
+                })
+            }
+        }
+        fetchData()
+    },[])
+
+    if(state.error){
+        return <Error/>
+    }
+    else if(state.loading){
+        return <Loading/>
+    }
     return(
         <>
-            <div className={styling.container}>
-                <img id={styling.panel} src={panelImg}/>
-                <div className={styling.panelText}>
-                    <h3>Join LearningHub to manage or inspire your students effortlessly or sign up and start learning the fundamentals of coding </h3>
+        <div className={styling.panelContainer}>
+            <img className={styling.panel} src={panelImg}/>
+            <div className={styling.panelText}>
+                <h3>Welcome to LearningHub: Your go-to platform for managing and inspiring students or diving into a wide range of educational materials. Whether you're an educator looking to streamline your classroom or a learner eager to explore new topics, join us today to enhance your teaching experience or expand your knowledge across various subjects.</h3>
+            </div>
+        </div>
+        <div className={styling.mainContainer}>
+            <h3>Checkout our available courses:</h3>
+                <div className={styling.coursesContainer}>
+                    {courseDescriptions.map((course,index) => (
+                        <div className={styling.courseContainer} key={index}>
+                            <img className={styling.courseLogo} src={course.imageUrl} alt="Course Image"/>
+                            <div className={styling.courseDescription}>
+                                <h1>{course.courseName}</h1>
+                                <p>Description:{course.description}</p>
+                                <p>Rating:{course.rating}</p>
+                            </div>
+                        </div>
+                    ))}
                 </div>
-            </div>     
+        </div>
         </>
+    )        
+}
+export default FrontPage;
+
+
+const Error = () =>{
+    return(
+        <div className={styling.errorContainer}>
+            <img src={pencil} className={styling.pencilImg}/>
+            <h1>Opps, something went wrong!</h1>
+            <p>We've encountered an error while trying to display our current available courses. Please wait a moment and try again.</p>
+        </div>
     )
 }
-export default FrontPage;
\ No newline at end of file
+
+const Loading = () => {
+    return(
+        <>
+        <h1>Loading...</h1>
+        <div className={styling.ldsspinner}><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div></div>
+        <p>Thank you for your patience.</p>
+        </>
+    )
+}
```

Comment:
I believe all `const` should be at the top of the file

Reply:
While this is true for small items like string literals and objects, it's not as clear for react components which is easier to read, and putting small sub-components beneath the main one is probably the more common paradigm. So I would leave for now. But I love the comments and that you're thinking about this!

(down the line, best practice with something like a spinner would be to make it a reusable component in the components folder, but that doesn't need to happen right now.)